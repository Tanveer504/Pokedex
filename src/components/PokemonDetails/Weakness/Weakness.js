import React from "react";
import "./Weakness.scss";
function Weakness({ weakness }) {
  return (
    <div className="pokemondetail__weakness">
      {weakness.map((weak, index) => (
        <span className={weak.name} key={index}>
          {weak.name}
        </span>
      ))}
    </div>
  );
}

export default Weakness;