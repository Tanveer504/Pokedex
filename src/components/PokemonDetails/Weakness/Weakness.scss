//Scss Beautifier
//mobile first approach
//weakness

@import "../../../styles/Breakpoint.scss";

//600 px below for mobile
@import '../../../styles/color.scss';
.pokemondetail__weakness {
  box-sizing: border-box;
  position: absolute;
  width: 70px;
  height: 24px;
  left: 31px;
  top: 766px;
  span {
    border: 1px solid #2e3156;
    border-radius: 5px;
    padding: 1px;
    text-transform: capitalize;
  }

  .ghost {
    background: $background-color_1;
  }
  .electric {
    background: $background-color_2;
  }
  .bug {
    background: $background-color_3;
  }
  .poison {
    background: $background-color_4;
  }
  .normal {
    background: $background-color_5;
  }
  .flying {
    background: $background-color_6;
  }
  .fairy {
    background: $background-color_7;
  }
  .fire {
    background: $background-color_8;
  }
  .grass {
    background: $background-color_9;
  }
  .water {
    background: $background-color_10;
  }
  .ground {
    background: $background-color_11;
  }
  .rock {
    background: $background-color_12;
  }
  .psychic {
    background: $background-color_13;
  }
}
@media only screen and (min-width: $min-width-desktop) {
  .pokemondetail__weakness {
    box-sizing: border-box;
    position: absolute;
    width: 70px;
    height: 24px;
    left: 430px;
    top: 500px;
    span {
      border: 1px solid #2e3156;
      border-radius: 5px;
      margin-right: 6px;
      padding: 1px;
    }
  }
}
