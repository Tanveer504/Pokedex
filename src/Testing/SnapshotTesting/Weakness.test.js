import renderer from 'react-test-renderer';
import Weakness from '../../components/PokemonDetails/Weakness'

it('should have the weakness component snapshot', () => {
    const tree = renderer.create(<Weakness />).toJSON();
    expect(tree).toMatchSnapshot();
});