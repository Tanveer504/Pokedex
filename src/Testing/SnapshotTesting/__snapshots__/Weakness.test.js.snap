// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should have the weakness component snapshot 1`] = `
<div
  className="PokemonDetail__Types"
/>
`;
